src/sentry/snuba/metrics/extraction.py

```diff
@@ -884,7 +886,10 @@ def _remove_blacklisted_search_filters(tokens: Sequence[QueryToken]) -> Sequence
     ret_val: List[QueryToken] = []
     for token in tokens:
         if isinstance(token, SearchFilter):
-            if token.key.name not in _IGNORED_METRIC_FIELDS:
+            if (
+                token.key.name not in _IGNORED_METRIC_FIELDS
+                and str(token) not in _IGNORED_METRIC_CONDITION
```

Comment:
I would add a test that tests that it ignores transaction (if we don't already have it) and not something else, so that we cover ourselves from regressions in case someone will ever modify the conditions string representation.

Reply:
The `test_cleanup_with_environment_injection` tests that `event.type:transaction` is ignored.

I also added a new test that verifies that `event.type:transaction` is completely ignored, otherwise, we would get this condition: `'condition': {'name': 'event.tags.event.type', 'op': 'eq', 'value': 'transaction'}`